spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/subscriber/ConvertedMessageReceiver.java

```diff
@@ -0,0 +1,30 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.pubsub.core.subscriber;
+
+import java.util.Map;
+
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+
+/**
+ * @author Mike Eltsufin
+ *
+ * @since 1.1
+ */
+public interface ConvertedMessageReceiver<T> {
+	void receiveMessage(T payload, Map<String, Object> headers, AckReplyConsumer ackReplyConsumer);
```

Comment:
We can potentially replace this with a plain old `MessageHandler` if the `AckReplyConsumer` is stored in the headers.